mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -5706,3 +5706,95 @@ fn test_electrum_v14_block_hash() {
+fn test_scan_and_deserialize_block_headers() {
+    // ========================== CONFIGURATION ==========================
+    /// The ticker of the coin to test (e.g., "NMC", "CHTA", "RVN").
+    const COIN_TICKER: &str = "NMC";
+    /// A list of active Electrum servers for the specified coin.
+    const ELECTRUM_URLS: &[&str] = &["nmc2.bitcoins.sk:57001", "nmc2.bitcoins.sk:57002"];
+    /// The block height to start scanning from.
+    const START_HEIGHT: u64 = 701614;
+    /// The block height to stop scanning at. Set to `None` to scan to the tip of the chain.
+    const END_HEIGHT: Option<u64> = Some(701616);
+    /// The number of headers to fetch in a single RPC call.
+    const CHUNK_SIZE: u64 = 100;
+    // ===================================================================
+
+    let client = electrum_client_for_test(ELECTRUM_URLS);
+    let mut current_height = START_HEIGHT;
+
+    loop {
+        let mut num_to_fetch = CHUNK_SIZE;
+        if let Some(end_h) = END_HEIGHT {
+            if current_height > end_h {
+                println!("Reached configured end height of {end_h}. Scan complete.");
+                break;
+            }
+            let remaining = end_h.saturating_sub(current_height) + 1;
+            num_to_fetch = num_to_fetch.min(remaining);
+        }
+
+        if num_to_fetch == 0 {
+            break;
+        }
+
+        println!("Fetching {num_to_fetch} headers from height {current_height}");
+        let headers_res =
+            block_on_f01(client.blockchain_block_headers(current_height, NonZeroU64::new(num_to_fetch).unwrap()))
+                .expect("Failed to get block headers");
+
+        if headers_res.count == 0 {
+            println!("Reached the end of the chain. No bad header found.");
+            break;
+        }
+
+        // This is the correct approach, inspired by your original test.
+        // We create a single reader for the entire raw byte stream of concatenated headers.
+        let raw_chunk_bytes = &headers_res.hex.0;
+        let mut reader = Reader::new_with_coin_variant(raw_chunk_bytes, COIN_TICKER.into());
+
+        // We loop exactly `count` times, reading one header in each iteration.
+        // The `read` method will correctly consume a variable number of bytes depending on the header's content.
+        for i in 0..headers_res.count {
+            let block_height_of_header = current_height + i;
+
+            if let Err(e) = reader.read::<BlockHeader>() {
+                // If a read fails, we've found the problematic header.
+                // We panic with all the necessary context for debugging.
+                let chunk_hex_str = hex::encode(raw_chunk_bytes);
+                panic!(
+                    "\n\n!!! Deserialization failed on header index {} (block height: {}) within the chunk starting at {} !!!\n\
+                     Raw Chunk Hex: {}\n\
+                     Deserialization Error: {:?}\n\n",
```

Comment:
format-wise, i think we should remove the space before `Raw Chunk ..` & `Deserialization Error..` so they don't get extra indented in the terminal.
i think we should also print the deserialization error first as the chunk_hex will probably be so big.